rpcserver.go

```diff
@@ -359,6 +361,42 @@ func handleAskWallet(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (
 	return nil, ErrRPCNoWallet
 }
 
+// handleAddMiningAddr implements the addminingaddr command.
+func handleAddMiningAddr(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.AddMiningAddrCmd)
+
+	// Decode the addresses encoded as a string into an in-memory object.
+	addr, err := btcutil.DecodeAddress(c.Address, activeNetParams.Params)
+	if err != nil {
+		return nil, &btcjson.RPCError{
+			Code:    btcjson.ErrRPCInvalidAddressOrKey,
+			Message: "Invalid address or key: " + err.Error(),
+		}
+	}
+
+	// Ensure the passed address belongs to the same network that btcd
+	// is currently on. Otherwise, this could lead to generating invalid
+	// block templates.
+	if !addr.IsForNet(activeNetParams.Params) {
+		return nil, &btcjson.RPCError{
+			Code: btcjson.ErrRPCInvalidAddressOrKey,
+			Message: fmt.Sprintf("Mining address %v is on the "+
+				"wrong network: ", activeNetParams.Params),
```

Comment:
Wouldn't this just substitute the `%v` with `activeNetParams.Params`? Maybe the line should be `fmt.Sprintf("Mining address %s is on the wrong network: %s", addr.String(), activeNetParams.Params.Name)` or something similar.